Reqnroll/BoDi/ObjectContainer.cs

```diff
@@ -364,7 +364,7 @@ public IStrategyRegistration RegisterTypeAs<TType, TInterface>(string name = nul
     public IStrategyRegistration RegisterTypeAs(Type implementationType, Type interfaceType, string name = null)
     {
         if (!IsValidTypeMapping(implementationType, interfaceType))
-            throw new InvalidOperationException("type mapping is not valid");
+            throw new InvalidOperationException($"type mapping is not valid: {implementationType} -> {interfaceType}");
```

Comment:
This error message is quite technical. Consider rewording it to something like `The provided type mapping is not valid. 
The {implementationType} is not assignable to {interfaceType}`

Reply:
Agreed. Modified.